### ISSUE TITLE
COS Payment Automation - TURP & EX New Permit Fees (On PROD)🎉🎉

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#7162]
### As the AMANDA system, I need to automatically pay permit and extension fees for our partner departments, using the Cost of Service payment type.

-----
### AC Clarification 10/20/21

On the COS Payment:
- The stamp user should be AMANDA
- The paidby rsn should be the folder applicant RSN
----

### New Permit Fees
**Acceptance Criteria**
**EX Folder**
- Folder types where this logic needs to be applied: All EX folders 
- Subtypes for EX Folders where this logic needs to be applied (#6322) :
-- Austin Energy
-- ATD
-- Austin Water
-- Corridor Program Office
-- Public Works
-- Other City Department
-- Watershed Protections
- When Info field 'Secondary?' (#6321) is answered 'No', System should automatically pay Excavation Permit Fee (#6844) that was inserted in the folder. 
-- Payment type should be recorded as 'Cost of Service' (this payment type has already been configured)
-- Paid by should be recorded as AMANDA (or any system account. Paid by should not contain actual user account IDs). 

**RW Folder**

- Folder types where this logic needs to be applied: RW folders with Temporary right of way use Subtype Only 
- Worktypes for RW-TURP Folders where this logic needs to be applied (#6322) :
-- Austin Energy
-- ATD
-- Austin Water
-- Corridor Program Office
-- Public Works
-- Other City Department
-- Watershed Protections
- System should automatically pay TURP New Permit Fee (#6844) that was inserted in the folder. 
-- Payment type should be recorded as 'Cost of Service' (this payment type has already been configured)
-- Paid by should be recorded as AMANDA (or any system account. Paid by should not contain actual user account IDs). 

-----

**Comment by TracyLinder** [comment]
Question for Paloma: Will our partner departments pay permit extension fees?

**Comment by vschandramouli** [comment]
Updated AC, pending review.

**Comment by TracyLinder** [comment]
Next Questions:
1. How do we automate the payment of extension fees for our partner departments on the TURP permit? 
2. Can we automate the payment of extension fees for EX? The users are manually inserting & billing these on EX,

**Comment by TracyLinder** [comment]
Convo w/ Paloma:

Q: Which extension fee should secondary permits pay? 
- Same cost as permit fee ($95)

**Comment by TracyLinder** [comment]
## Scoping Sync w/ Nate & Vish

**Breaking up this into development chunks (Epic)**
_Ordered by priority_
1. TURP & EX (New Permit Fees)
2. UC Review Fee
3. TURP Extension Fees
4. EX extension fees
5. TCP fee on GF folder

**Comment by TracyLinder** [comment]
@vschandramouli I'm not sure if this fits in this issue, but we need to capture this somewhere:

- COS permits need to issue themselves after the COS payment is made.

**Comment by TracyLinder** [comment]
Vish should have the AC ready for this by end of today.

**Comment by SubhaShroff** [comment]
@TracyLinder  @Nadin-Nader  @vschandramouli 

- Tracy suggested to go for GL accountnumber for COS as 1. - Done.
- As per acceptance criteria Paid By should be 'AMANDA' or any system account. The paid by column takes a peopleRSN. Is there any already configured people record for internal payment purpose? Or one need to be created?
- Though it is not a mandatory field, do you want any particular text go in Memo field? 

This is JV payment 
![image](https://user-images.githubusercontent.com/51680715/137365163-bc859f65-e301-4195-93bc-35d71eae3e63.png)

COS Payment without Paid By & Memo:
![image](https://user-images.githubusercontent.com/51680715/137365943-405f628b-655a-4f6d-9848-c6be40426e08.png)

**Comment by SubhaShroff** [comment]
@TracyLinder  @Nadin-Nader  @vschandramouli  
A people record is created as follows. Please let me know if you want any changes or additions to it.
![image](https://user-images.githubusercontent.com/51680715/137522694-049353aa-932d-4d02-889f-c98ad1156ae1.png)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  @vschandramouli  SCTASK0174388 is submitted for DEV deployment, after Nadin gave approval on Teams.

![image](https://user-images.githubusercontent.com/51680715/137939320-42919034-b806-49cf-817d-16a326ccf570.png)

**Comment by Nadin-Nader** [comment]
@SubhaShroff can you send me a demo of this on DEV? My Dev portal is not allowing me to log in.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Sent you demo through email, today.

**Comment by Nadin-Nader** [comment]
@TracyLinder; @SubhaShroff the demos look good. In my opinion, this can go to **TEST**

**Comment by SubhaShroff** [comment]
@TracyLinder  @Nadin-Nader  SCTASK0177830 - is submitted for TEST deployment.

**Comment by SubhaShroff** [comment]
### Status:
@TracyLinder  @Nadin-Nader  Tested for it on TEST and its ready for your testing.

**Comment by Nadin-Nader** [comment]
**Status:**
Sent Paloma a message via Team Chat asking her to UAT this when she has a moment so we can get this on PROD. Waiting response

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  After UATing today,
Other City Department EX, RW-TURP also should be COS Auto-payment.  I am going to add Other City Department  to the list that goes in COS Auto-payment.   Project Connect  is already taken care.

@vschandramouli  Can you please update the AC for the below or if I am allowed to edit AC, I can do that 🙂 :
- Project Connect -  **NO** COS Auto-payment.
- Other City Department - Add to COS Auto-payment.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  SCTASK0179303 is submitted for DEV deployment to add Other City Department to COS Auto-payment.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  I tested the DEV deployment for Other City Department from DEV Portal -

![image](https://user-images.githubusercontent.com/51680715/140972001-ec6ece66-b1f1-4481-815a-8bd320a70d97.png)

![image](https://user-images.githubusercontent.com/51680715/140973098-d3a00869-5eec-4f9d-8f70-f3ddcfaed5d7.png)

**Comment by SubhaShroff** [comment]
### Status:
Waiting on Batch related Deployment in DEV.

**Comment by TracyLinder** [comment]
**Feedback from me**
- Both Paid By & Stamp User should be AMANDA

**Comment by TracyLinder** [comment]
**Update:** Subha is Working on the commit for new changes going to DEV

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  SCTASK0180230 is submitted for DEV/TEST deployment of the commit to change Paid-by as 'AMANDA'.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder   is causing some compilation errors in DEV portal, so I had to get the latest commit reverted. SCTASK0180545 is submitted to revert.
I will debug and make a fresh commit.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0180593 is submitted to deploy ATD COS AMANDA Payer people record to DEV.  It may need your comment in the ticket as deploying people record is not common.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The manual deployment is done. So, I submitted SCTASK0181233 for DEV deployment of code changes for Paid-by change.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to **_DEV_** please UAT

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  The change is deployed in DEV. I tested it on Portal as well.

![image](https://user-images.githubusercontent.com/51680715/142659126-bf5194aa-2fd4-4bcf-8c03-9783228766b8.png)

![image](https://user-images.githubusercontent.com/51680715/142659482-e941add7-4ff2-4b7e-b3c2-34afb2a89f44.png)

**Comment by SubhaShroff** [comment]
@Nadin-Nader 
 SCTASK0181404  TEST (for commit deployment in TEST), 
SCTASK0181398 (submitted for People record manual deployment in TEST) are raised.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder 
Tested on TEST. The AC is fulfilled but I have noticed the following issue which is not in this ticket's scope (the issue is mentioned in the 2nd bullet point).

- On Friday 2pm call, Paloma had mentioned - if the permits are saved not submitted the status should be 'Application Incomplete' and it should change to 'Internet Pending' with submission. This change is also implemented.
- While testing the above scenario, I noticed, the fees are getting added to the permit with every 'Save it for Later' and with Submission as well. @Sreelatha-BlazeKannan  I am not sure about the ticket number(6844?) that related to Fee addition, you need to restrict the fee addition. You may need to add a check if the fee is already added.
Ref: 2021-048737 RW on TEST.

**Comment by SubhaShroff** [comment]
### Status:

- REady for test on TEST.
- A bug is found about adding fee, though its out of scope for this ticket. However, I informed this to Sree/Nadin.

**Comment by SubhaShroff** [comment]
@TracyLinder  @Nadin-Nader  Today in a call with Kim, suggested the following change:
Permit New fees can be added to incomplete applications. But, fees get billed and the COS Automation Payment will happen only with Portal submission.

So, I will do the change as per the suggestion.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0183181 is requested for DEV deployment of the commit done for the change: Permit fees can be billed and COS auto-paid with only Portal submission.

**Comment by SubhaShroff** [comment]
### Status:
The DEV deployment of the change is done. I need to test it.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  tested the new change - to bill and do autopayment only with SUBMIT button. It looks good. Please let me know if I can deploy in TEST.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  on your teams approval, I have submitted SCTASK0183537 for TEST Deployment of the above change.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The ticket is deployed with latest change in TEST. I tested for the following, and its ready for UAT.
2021-048986 EX - Watershed Protection Department WPD - as expected.  The same will follow with
2021-048987 RW - Project Connect - No automation payment happened. 
--- Yet to be tested.

**Comment by Nadin-Nader** [comment]
**Status:**
DS side is not working @SubhaShroff is currently debugging

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on #7707 in order to go to PROD

**Comment by TracyLinder** [comment]
**UAT Note:**

I tested an EX with 2 streets, ATD subtype. The system did not insert fees on both folders. This is a bug being fixed on #7352. 


<img width="695" alt="Screen Shot 2021-12-10 at 2 47 35 PM" src="https://user-images.githubusercontent.com/49930868/145639440-a5a13ec0-a26b-4a4f-bcf0-3ae2eadf6583.png">

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0185962 is submitted for the manual deployment of People record - ATD COS Amanda Payer'

**Comment by Nadin-Nader** [comment]
@SubhaShroff @TracyLinder I have sent this to the DevOps CM Team to deploy to **_PROD_** under **SCTASK0186329**

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on  the DevOps CM Team to deploy to **_PROD_** under **SCTASK0186329**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder This ticket is in PROD now.

**Comment by SubhaShroff** [comment]
### Status:
Deployed 4 days ago. Waiting to close.

**Comment by Nadin-Nader** [comment]
Closing as i have not heard anything on this, it is on PROD and i informed the stakeholders yesterday that i would be closing it
